modules/local/sum_eggnog.nf

```diff
@@ -0,0 +1,64 @@
+process SUM_EGGNOG {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "conda-forge::r-tidyverse=1.3.1 conda-forge::r-data.table=1.14.0 conda-forge::r-dtplyr=1.1.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    'https://depot.galaxyproject.org/singularity/mulled-v2-508c9bc5e929a77a9708902b1deca248c0c84689:0bb5bee2557136d28549f41d3faa08485e967aa1-0' :
+    'quay.io/biocontainers/mulled-v2-508c9bc5e929a77a9708902b1deca248c0c84689:0bb5bee2557136d28549f41d3faa08485e967aa1-0' } "
+
+    input:
+
+    tuple val(meta), path(eggnog)
+    path(fcs)
+
+    output:
+
+    path "eggnog_summary.tsv" , emit: eggnog_summary
+    path "versions.yml"       , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    #!/usr/bin/env Rscript
+
+    library(data.table)
+    library(dtplyr)
```

Comment:
I can't see that you're using neither of these two packages. Maybe it would make the code faster if you did though. If so, you need to make sure all tables are data.table and end each pipeline with a `as.data.table()`.